cluster/gce/util.sh

```diff
@@ -132,6 +132,26 @@ function detect-project () {
   fi
 }
 
+function sha1sum-file() {
+  if which shasum >/dev/null 2>&1; then
+    shasum -a1 "$1" | awk '{ print $1 }'
+  else
+    sha1sum "$1" | awk '{ print $1 }'
+  fi
+}
+
+function already-staged() {
+  local -r file=$1
+  local -r newsum=$2
+
+  if [ ! -e "${file}.sha1" ]; then
+    return 1
+  fi
+
+  oldsum=$(cat "${file}.sha1")
```

Comment:
oldsum should be local.

There's a problem with `local v=$(...)` in that the `local` statement will hide an error of the command in the `$(...)` expression, so declare it in a separate line.

```shell
  local oldsum
  oldsum=$(cat "${file}.sha1")
```